File: example/Calculator.java
/**
 * A simple calculator class demonstrating Java support
 */
public class Calculator {
    
    /**
     * Adds two numbers together
     * @param a First number
     * @param b Second number
     * @return Sum of a and b
     */
    public int add(int a, int b) {
        return a + b;
    }
    
    /**
     * Subtracts second number from first
     * @param a First number
     * @param b Second number
     * @return Difference of a and b
     */
    public int subtract(int a, int b) {
        return a - b;
    }
    
    /**
     * Multiplies two numbers
     * @param a First number
     * @param b Second number
     * @return Product of a and b
     */
    public int multiply(int a, int b) {
        return a * b;
    }
    
    /**
     * Divides first number by second
     * @param a Dividend
     * @param b Divisor
     * @return Quotient of a and b
     * @throws ArithmeticException When dividing by zero
     */
    public int divide(int a, int b) throws ArithmeticException {
        if (b == 0) {
            throw new ArithmeticException("Division by zero");
        }
        return a / b;
    }
    
    /**
     * Calculates the power of a number
     * @param base Base number
     * @param exponent Exponent
     * @return base raised to the power of exponent
     */
    public double power(double base, int exponent) {
        return Math.pow(base, exponent);
    }
    
    /**
     * Main method to demonstrate calculator usage
     */
    public static void main(String[] args) {
        Calculator calc = new Calculator();
        
        System.out.println("Addition: 5 + 3 = " + calc.add(5, 3));
        System.out.println("Subtraction: 10 - 4 = " + calc.subtract(10, 4));
        System.out.println("Multiplication: 6 * 7 = " + calc.multiply(6, 7));
        System.out.println("Division: 15 / 3 = " + calc.divide(15, 3));
        System.out.println("Power: 2^8 = " + calc.power(2, 8));
    }
}

File: example/Calculator.cs
/**
 * A simple calculator class demonstrating C# support
 */
using System;

public class Calculator {
    
    /**
     * Adds two numbers together
     */
    public int Add(int a, int b) {
        return a + b;
    }
    
    /**
     * Subtracts second number from first
     */
    public int Subtract(int a, int b) {
        return a - b;
    }
    
    /**
     * Multiplies two numbers
     */
    public int Multiply(int a, int b) {
        return a * b;
    }
    
    /**
     * Divides first number by second
     */
    public double Divide(double a, double b) {
        if (b == 0) {
            throw new DivideByZeroException("Division by zero");
        }
        return a / b;
    }
    
    /**
     * Calculates the power of a number
     */
    public double Power(double baseNum, int exponent) {
        return Math.Pow(baseNum, exponent);
    }
    
    /**
     * Main method to demonstrate calculator usage
     */
    public static void Main(string[] args) {
        Calculator calc = new Calculator();
        
        Console.WriteLine("Addition: 5 + 3 = " + calc.Add(5, 3));
        Console.WriteLine("Subtraction: 10 - 4 = " + calc.Subtract(10, 4));
        Console.WriteLine("Multiplication: 6 * 7 = " + calc.Multiply(6, 7));
        Console.WriteLine("Division: 15 / 3 = " + calc.Divide(15.0, 3.0));
        Console.WriteLine("Power: 2^8 = " + calc.Power(2, 8));
    }
}